data_safe_haven/commands/shm.py

```diff
@@ -142,6 +142,14 @@ def teardown() -> None:
     try:
         config = SHMConfig.from_remote(context)
         shm_infra = ImperativeSHM(context, config)
+        console.print(
+            "Tearing down the Safe Haven Management environment will permanently delete all associated resources, including remotely stored configurations."
+        )
+        if not console.confirm(
+            "Do you wish to continue tearing down the SHM?", default_to_yes=True
+        ):
+            logger.info("SHM teardown cancelled by user.")
```

Comment:
```suggestion
```
I think we shouldn't need this because the `console.confirm()` method adds to the logs.

Reply:
I think it's still useful to confirm it for the user, so changed these to console.print